org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/thingTranslator.js

```diff
@@ -120,69 +128,95 @@ class WinkThermostat {
     }
 
-    // exports for the OCF schema
+    // exports for the entire schema object
 
     // Queries the entire state of the thermostat
     // and returns an object that maps to the json schema org.opent2t.sample.thermostat.superpopular
     getThermostatResURI() {
         return winkHelper.getDeviceDetailsAsync(deviceType, deviceId)
             .then((response) => {
-                return winkSchemaToOcfSchema(response.data);
+                return deviceSchemaToTranslatorSchema(response.data);
             });
     }
 
-    // Updates the current state of the thermostat with the contents of value
-    // value is an object that maps to the json schema org.opent2t.sample.thermostat.superpopular
+    // Updates the current state of the thermostat with the contents of postPayload
+    // postPayload is an object that maps to the json schema org.opent2t.sample.thermostat.superpopular
     //
     // In addition, returns the updated state (see sample in RAML)
-    postThermostatResURI(value) {
-        var putPayload = ocfSchemaToWinkSchema(value);
+    postThermostatResURI(postPayload) {
+        console.log('postThermostatResURI called with payload: ' + JSON.stringify(postPayload));
+
+        var putPayload = translatorSchemaToDeviceSchema(postPayload);
         return winkHelper.putDeviceDetailsAsync(deviceType, deviceId, putPayload)
             .then((response) => {
-                return winkSchemaToOcfSchema(response.data);
+                return deviceSchemaToTranslatorSchema(response.data);
             });
     }
 
-    // exports for the AllJoyn schema
+    // exports for individual properties
+
     getAmbientTemperature() {
         console.log('getAmbientTemperature called');
-        return winkHelper.getLastReadingAsync(deviceType, deviceId, 'temperature');
+
+        return winkHelper.getDeviceDetailsAsync(deviceType, deviceId)
+            .then((response) => {
+                return deviceSchemaToTranslatorSchema(response.data).ambientTemperature.temperature;
+            });
     }
 
     getTargetTemperature() {
         console.log('getTargetTemperature called');
 
-        // Wink does not have a target temperature field, so returning the average of min and max setpoint
-        return this.getTargetTemperatureHigh()
-            .then(high => {
-                return this.getTargetTemperatureLow()
-                    .then(low => {
-                        return (high + low) / 2;
-                    });
+        return winkHelper.getDeviceDetailsAsync(deviceType, deviceId)
+            .then((response) => {
+                return deviceSchemaToTranslatorSchema(response.data).targetTemperature.temperature;
             });
     }
 
     getTargetTemperatureHigh() {
         console.log('getTargetTemperatureHigh called');
 
-        return winkHelper.getLastReadingAsync(deviceType, deviceId, 'max_set_point');
+        return winkHelper.getDeviceDetailsAsync(deviceType, deviceId)
```

Comment:
>winkHelper.getDeviceDetailsAsyn [](start = 15, length = 31)

why not call getThermostateResURI directly? that way you won't have to call convert schemas here, it'll be a simpler: 

return response.targetTemperature.temperature